### PR TITLE
Setup sign and publishing configuration

### PR DESCRIPTION
**Pull request by e5l**
Setup GitHub action and gradle configuration to sign publish artifacts

### CODE REVIEW

File: .github/workflows/gradle-publish.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+# This workflow will build a package using Gradle and then publish it to GitHub packages when a release is created
+# For more information see: https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#Publishing-using-gradle
+
+name: Gradle Package
```
Comment (Mr3zee) [inline .github/workflows/gradle-publish.yml]: `Release`?

File: .github/workflows/gradle-publish.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK 21
+        uses: actions/setup-java@v4
+        with:
+          java-version: '21'
+          distribution: 'temurin'
+          server-id: ossrh # Value of the distributionManagement/repository/id field of the pom.xml
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
```
Comment (Mr3zee) [inline .github/workflows/gradle-publish.yml]: `OSSRH_USERNAME` and `OSSRH_TOKEN`

Comment (e5l) [reply .github/workflows/gradle-publish.yml]: please check: https://docs.github.com/en/actions/use-cases-and-examples/publishing-packages/publishing-java-packages-with-maven#publishing-packages-to-the-maven-central-repository

File: .github/workflows/gradle-publish.yml (hunk 1 above)
Comment (e5l) [reply .github/workflows/gradle-publish.yml]: thank you, fixed

**Comment by e5l** [comment]
Hey, @Mr3zee! Thank you for the review, pushed update

File: .github/workflows/gradle-publish.yml (hunk 2 above)
Comment (Mr3zee) [reply .github/workflows/gradle-publish.yml]: Yes, but you have `OSSRH_USERNAME` and `OSSRH_TOKEN` defined in gradle configs and in secrets mapping in this yaml below

Comment (Mr3zee) [reply .github/workflows/gradle-publish.yml]:
```yaml
env:
  OSSRH_USERNAME: ${{ secrets.OSSRH_USERNAME }}
  OSSRH_TOKEN: ${{ secrets.OSSRH_TOKEN }}
```

Comment (Mr3zee) [reply .github/workflows/gradle-publish.yml]:
```kotlin
val ossrhUsername = System.getenv("OSSRH_USERNAME")
    ?: project.findProperty("kotlin.mcp.sdk.packages.username") as String?
val ossrhPassword = System.getenv("OSSRH_TOKEN")
    ?: project.findProperty("kotlin.mcp.sdk.packages.password") as String?
```

Comment (Mr3zee) [reply .github/workflows/gradle-publish.yml]: They should be unified to use one name, I just think `OSSRH_` ones are better

Comment (e5l) [reply .github/workflows/gradle-publish.yml]: Thanks, aligned!

File: .github/workflows/gradle-publish.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK 21
+        uses: actions/setup-java@v4
+        with:
+          java-version: '21'
+          distribution: 'temurin'
+          server-id: ossrh # Value of the distributionManagement/repository/id field of the pom.xml
+          server-username: OSSRH_USERNAME
+          server-password: OSSRH_TOKEN
```
Comment (dsp-ant) [inline .github/workflows/gradle-publish.yml:32]: Will this read the username and token already or just insert it into the pom.xml?

File: .github/workflows/gradle-publish.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        run: ./gradlew build
+
+      - name: Publish to GitHub Packages
+        run: ./gradlew publish
+        env:
+          OSSRH_USERNAME: ${{ secrets.OSSRH_USERNAME }}
+          OSSRH_TOKEN: ${{ secrets.OSSRH_TOKEN }}
+          SIGNING_KEY_ID: ${{ secrets.SIGNING_KEY_ID }}
+          SIGNING_KEY_PRIVATE: ${{ secrets.SIGNING_KEY_PRIVATE }}
+          SIGNING_PASSPHRASE: ${{ secrets.SIGNING_PASSPHRASE }}
```
Comment (dsp-ant) [inline .github/workflows/gradle-publish.yml]: If only this step needs the secrets, than we should do this:

```suggestion
      - name: Publish to GitHub Packages
        run: ./gradlew publish
        environment: release
        env:
          OSSRH_USERNAME: ${{ secrets.OSSRH_USERNAME }}
          OSSRH_TOKEN: ${{ secrets.OSSRH_TOKEN }}
          SIGNING_KEY_ID: ${{ secrets.SIGNING_KEY_ID }}
          SIGNING_KEY_PRIVATE: ${{ secrets.SIGNING_KEY_PRIVATE }}
          SIGNING_PASSPHRASE: ${{ secrets.SIGNING_PASSPHRASE }}
```

See https://github.com/modelcontextprotocol/python-sdk/blob/main/.github/workflows/publish-pypi.yml where we only call it in the release step. This way only the actually publishing requires a review.

File: .github/workflows/gradle-publish.yml (hunk 3 above)
Comment (e5l) [reply .github/workflows/gradle-publish.yml:32]: It will read it from the env section below

File: .github/workflows/gradle-publish.yml (hunk 4 above)
Comment (e5l) [reply .github/workflows/gradle-publish.yml]: sure, let me adjust it

Comment (e5l) [reply .github/workflows/gradle-publish.yml]: The schema suggests that the property is forbidden inside the step:
image

Comment (dsp-ant) [reply .github/workflows/gradle-publish.yml]: Ahh because this is a step. We have different "jobs" that depend on each other in the pypi.yml.

**Pull request merged**